### PR TITLE
Redundant check for empty solutionFilePath fix

### PR DESCRIPTION
**Pull request by JDemler**

### CODE REVIEW

**Comment by davkean** [comment]
@dotnet-bot okay to test

**Comment by davkean** [comment]
@dotnet-bot test this please

**Comment by davkean** [comment]
Tag @dotnet/roslyn-ide, please look at this

**Comment by davkean** [comment]
:+1:

**Comment by brettfo** [comment]
Only tangentially related to this PR, but do the hard-coded back slashes need to be replaced with [`Path.DirectorySeparatorChar`](https://msdn.microsoft.com/en-us/library/system.io.path.directoryseparatorchar(v=vs.110).aspx), or does the `MSBuildWorkspace` normalize this later?  If the hard-coded backslashes aren't taken care of, then this won't work on non-Windows or on Japanese builds.

**Comment by davkean** [comment]
@brettfo Back slashes on Japanese builds are still back slashes, they are just rendered differently as ¥.

**Comment by davkean** [comment]
Can I have one more thumbs up from the @dotnet/roslyn-ide team?

File: src/Workspaces/Core/Desktop/Workspace/MSBuild/MSBuildProjectLoader.cs [hunk 1]
```diff
@@ -97,25 +97,22 @@ public void AssociateFileExtensionWithLanguage(string projectFileExtension, stri
...
-                        solutionDirectory += @"\";
-                    }
+                    solutionDirectory += @"\";
+                }
 
-                    if (Directory.Exists(solutionDirectory))
-                    {
-                        _properties = _properties.SetItem(SolutionDirProperty, solutionDirectory);
-                    }
+                if (Directory.Exists(solutionDirectory))
```
Comment (jasonmalinowski) [inline src/Workspaces/Core/Desktop/Workspace/MSBuild/MSBuildProjectLoader.cs]: Isn't the solution file at this point known to exist? (Aren't we already loading it?) Can this ever fail?

(yes, I know this logic was there, but if we're deleting stupid code we mind as well delete more stupid code)

**Comment by jasonmalinowski** [comment]
But my question aside, I'll give you two thumbs up! :+1: :+1:

**Comment by davkean** [comment]
@jasonmalinowski If you think the current behavior is bad with this method, can you file a bug, that may more downstream impacts that external contrib cannot probably answer or find out easily? 

I'm merging this in, thanks @JDemler for your contribution!

**Pull request merged**